src/magicgui/backends/_qtpy/widgets.py

```diff
@@ -908,7 +908,7 @@ def _mgui_bind_change_callback(self, callback):
         self._event_filter.valueChanged.connect(callback)
 
     def _mgui_get_count(self) -> int:
-        """Return the number of items in the dropdown."""
+        """Return the number of items in the dropdown, including separator items."""
```

Comment:
this actually becomes a bit problematic.  The protocol specifies that `_mgui_get_count` should return the number of _choices_:

https://github.com/pyapp-kit/magicgui/blob/781a54a3d0fb49845ededf734dbf30f71e18509c/src/magicgui/widgets/protocols.py#L412-L415

so, it's not really up to the backend here to change those semantics.   The `ComboBox` itself (via `CategoricalWidget`) declares `__len__` to be the number of choices (not the number of widgets)

https://github.com/pyapp-kit/magicgui/blob/781a54a3d0fb49845ededf734dbf30f71e18509c/src/magicgui/widgets/bases/_categorical_widget.py#L105-L107

so, `assert len(combo_a) == len(combo_a.choices)` should technically be true, but is not currently so in the tests.   So, I'm afraid this needs to account for that as well.  

I know these are annoying things, but this feature is a revealing itself to be a bit tricky to implement in a way that isn't potentially surprising and too magical.  It is blurring the lines between model (i.e. the values in `choices`) and the view (how they look in the combobox) a bit. 

i'm honestly not sure what the best thing to do here is.  one alternative is to go back to what you started with, and just allow for `choices` to include the separator.  that is, when you call `widget.choices` you *do* see all the `Separator` singletons in there (they would need to be subbed back in for `None`).   Or, we just fix the count issue as well and stick with it as it is.  That would then mean that it's basically impossible to query where the separators are, and whether they're in there (but you could also set them).  Which do you prefer?